### DISCUSSION TITLE
IS IT POSSIBLE TO ACCESS THE EBSD DATA (.ANG FILE) AND CREATE A MICROSTRUCTURE?

### DISCUSSION THREAD

**Discussion by Abhinab-iit-112** [neperfepx/neper#897] [General]
I want to use .ang file directly and create a micro-structure so that i can import it to abaqus...is it possible via any code ??
please do let me know
```
neper -T -n20 -id1 -eulerfile ebsd.ang -o mesh.inp

========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.9.1-2
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 4 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2022, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (/home/user1233/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -T loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] -n20 -id1 -eulerfile ebsd.ang -o mesh.inp
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Error  : Unknown option -n20'.
Error  : Bad arguments!

Aborted
error  coming
```

**Comment by Abaqus-Voronoi-toolbox** [comment]
See https://neper.info/doc/tutorials/ebsd_process.html
You should process your EBSD data to tesr file,  or see https://github.com/neperfepx/neper/discussions/258

**Reply by Abhinab-iit-112** [reply]
> See https://neper.info/doc/tutorials/ebsd_process.html You should process your EBSD data to tesr file, or see #258

Dear sir,
i am completely new to this 
and also it is showing an error
![Screenshot (290)](https://github.com/user-attachments/assets/d6caf968-6e76-4fc8-93b9-69f67a884c10)
![Uploading Screenshot (291).png…]()

**Comment by rquey** [comment]
There is no `-eulerfile` option. Use `-ori file(<file_name>,des=euler-bunge)`, as described [here](https://neper.info/doc/neper_t.html#cmdoption-ori). Also make sure to insert a space between `-n` and `20`. This will generate a `.tess` file that can then be meshed. Use `-format inp` to get a file at Abaqus format.

See [this tutorial](https://neper.info/doc/tutorials/simple_model.html).

Example commands:
```
neper -T -n 20 -ori "file(myfile,des=eurler-bunge)" -reg 1
neper -M n20-id1.tess -format inp
```
